.gitignore

```diff
@@ -61,8 +61,8 @@ autoconf/autom4te.cache
 .vs
 # pythonenv for github Codespaces
 pythonenv*
-# clangd index. (".clangd" is a config file now, thus trailing slash)
-.clangd/
+# clangd config file and index 
+.clangd
```

Comment:
Why do we need to make this change in intel/llvm repository rather than in llvm/llvm-project?

Reply:
ok, move this change to libdevice

Reply:
we can add config to ~/.config/clangd/config.yaml rather than adding .clangd file:
```
---

If:
  PathMatch: /.../llvm/.*

CompileFlags:
  CompilationDatabase: /.../llvm/build
  Add: [-D...]

Diagnostics:
  ClangTidy:
    Remove: readability-identifier-naming
```

Reply:
I agree with @wenju-he. Please, remove this file from your patch and use local config instead.

Reply:
Removed. 
Although I don't understand why you all dislike to introduce a more convenient way to configure clangd (directly configure you setting in the project folder).

Reply:
FYI you can add the ignore of `.clangd` to `.git/info/exclude` to only affect your local clone without setting these options globally in `~/.config/clangd/config.yaml`.

Reply:
The change has nothing to do with the "checking sycl::free related errors", so it should be not added to this PR.

According to my understanding it's not specific to SYCL project. If you want to make this change, our contribution guidelines say that it should be committed to llvm/llvm-project repository. https://github.com/intel/llvm/blob/sycl/CONTRIBUTING.md#development

Reply:
Thank you for explanation! Got it.